internal/sshserver/serverImpl.go

```diff
@@ -54,11 +55,20 @@ func (s *serverImpl) RunWithLifecycle(lifecycle service.Lifecycle) error {
 		Control: s.socketControl,
 	}
 
+	useProxy := len(s.cfg.AllowedProxies) > 0
+
 	netListener, err := listenConfig.Listen(lifecycle.Context(), "tcp", s.cfg.Listen)
 	if err != nil {
 		s.lock.Unlock()
 		return messageCodes.Wrap(err, messageCodes.ESSHStartFailed, "failed to start SSH server on %s", s.cfg.Listen)
 	}
+	if useProxy {
```

Comment:
It seems to me that PROXY is a wrapper protocol like many others (e.g. websocket). We should consider creating a generic abstraction for wrapper protocols.

Reply:
I don't foresee having any other wrapping protocols in-front of the ssh server since, we can consider that if it's necessary further down the line.